src/Corax/IndexFieldsMapping.cs

```diff
@@ -1,24 +1,28 @@
 ﻿using System;
 using System.Collections;
+using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.Diagnostics;
 using System.IO;
 using System.Runtime.InteropServices;
 using Sparrow.Server;
+using Sparrow.Threading;
 using Voron;
 
 namespace Corax;
 
-public abstract class IndexFieldsMappingBase : IEnumerable<IndexFieldBinding>
+public abstract class IndexFieldsMappingBase : IEnumerable<IndexFieldBinding>, IDisposable
 {
     protected const short LongSuffix = 19501; //"-L"
     protected const short DoubleSuffix = 17453; //"-D"
     
-    protected readonly ByteStringContext Context;
+    /// <summary>
+    /// Inner allocator to hold sliced names. Do not use it for querying.
+    /// </summary>
+    protected readonly ByteStringContext InnerContext;
     
-    protected readonly Dictionary<Slice, IndexFieldBinding> Fields;
-    protected readonly Dictionary<int, IndexFieldBinding> FieldsById;
-    protected readonly List<IndexFieldBinding> FieldsList;
+    protected readonly ConcurrentDictionary<Slice, IndexFieldBinding> Fields;
+    protected readonly ConcurrentDictionary<int, IndexFieldBinding> FieldsById;
     
```

Comment:
Why?
Are we actually _changing_ these fields after we create it?

It is fine to have a readonly dictionary in multiple threads